src/pykeen/quadruples/quadruples_factory.py

```diff
@@ -0,0 +1,1246 @@
+    if len(columns) > 1:
+        mask = mask.all(dim=-1)
+    return mask
+
+
+def restrict_quadruples(
+    mapped_quadruples: MappedQuadruples,
+    entities: Optional[Collection[int]] = None,
+    relations: Optional[Collection[int]] = None,
+    timestamps: Optional[Collection[int]] = None,
+    invert_entity_selection: bool = False,
+    invert_relation_selection: bool = False,
+    invert_timestamp_selection: bool = False,
+) -> MappedQuadruples:
+    """Select a subset of quadruples based on the given entity and relation ID selection.
+
+    :param mapped_quadruples:
+        The ID-based quadruples.
+    :param entities:
+        The entity IDs of interest. If None, defaults to all entities.
+    :param relations:
+        The relation IDs of interest. If None, defaults to all relations.
+    :param timestamps:
+        The relation IDs of interest. If None, defaults to all timestamps.
+    :param invert_entity_selection:
+        Whether to invert the entity selection, i.e. select those quadruples without the provided entities.
+    :param invert_relation_selection:
+        Whether to invert the relation selection, i.e. select those quadruples without the provided relations.
+    :param invert_timestamp_selection:
+        Whether to invert the timestamp selection, i.e. select those quadruples without the provided timestamps.
+    :return:
+        A tensor of quadruples containing the entities and relations of interest.
+    """
+    keep_mask = None
+
+    # Filter for entities
+    if entities is not None:
+        keep_mask = _get_quadruple_mask(
+            ids=entities,
+            quadruples=mapped_quadruples,
+            columns=(0, 2),  # head and entity need to fulfil the requirement
+            invert=invert_entity_selection,
+        )
+
+    # Filter for relations
+    if relations is not None:
+        relation_mask = _get_quadruple_mask(
+            ids=relations,
+            quadruples=mapped_quadruples,
+            columns=1,
+            invert=invert_relation_selection,
+        )
+        keep_mask = relation_mask if keep_mask is None else keep_mask & relation_mask
+
+    # Filter for timestamps
+    if timestamps is not None:
+        timestamp_mask = _get_quadruple_mask(
+            ids=timestamps,
+            quadruples=mapped_quadruples,
+            columns=3,
+            invert=invert_timestamp_selection,
+        )
+        keep_mask = timestamp_mask if keep_mask is None else keep_mask & timestamp_mask
+
+    # No filter
+    if keep_mask is None:
+        return mapped_quadruples
+
+    return mapped_quadruples[keep_mask]
+
+
+def create_timestamp_mapping(timestamps: set) -> TimestampMapping:
+    """Create sorted timestamp mapping."""
+    # format of the timestamp is YYYY-MM-DD.
+    timestamp_labels = sorted(set(timestamps), key=lambda date: datetime.strptime(date, "%Y-%m-%d"))
+
+    return {str(label): i for (i, label) in enumerate(timestamp_labels)}
+
+
+def _map_quadruples_elements_to_ids(
```

Comment:
can we merge this method with triple mapping / generalize to load n-ary-triples instead?

Reply:
Yes I have generalized this method by adding the parameter "is_triples" in the new commit